src/ol/worker/version.js

```diff
@@ -4,9 +4,14 @@
  */
 import {VERSION} from '../util.js';
 
-onmessage = event => {
+/**
+ * @type {Worker}
+ */
+const worker = /** @type {any} */ (self);
```

Comment:
The problem is that now `worker` type is `any` (at least for intellisense).
What about this: https://github.com/openlayers/openlayers/compare/master...fredj:webworker_jsconfig ?

It move the `webworker` lib only for the files inside the `src/ol/worker` directory

Reply:
Thanks! I've tried that but to no avail. Will check your branch & update the PR.

Reply:
not sure if it was clear but the commit in my branch replaces this one

Reply:
> It move the webworker lib only for the files inside the src/ol/worker directory

Did not work on my side. I've turned the problem around many times but just could not get tsc to use the webworker lib only for the `worker` folder, and the dom lib for the rest.. In the end I went with a manual extraction of the worker scope typings, see: https://github.com/openlayers/openlayers/pull/9710/commits/ae48809b252248bbc64f062faca90a7307921586

Hope that'll be fine, otherwise you're welcome to try other things on this branch ;)